### ISSUE TITLE
roadmap

### ISSUE THREAD

**Issue by wangjiawen2013** [yuankunzhang/charming#32]
Hi,

Thanks for your great work! I am looking for rust visualization library for a long time. I have tried plotters (https://github.com/plotters-rs/plotters) and plotly (https://github.com/igiagkiozis/plotly), both of them are disappointing cannot meet my demand. Charming is the best one of all the rust visualization library I have ever seen and worth trying!

Do you have a roadmap of charming and will you maintain this project ? Though I am not a developer, I can test charming for you. And can I convert charming chart to json now using something like Chart.to_json() ? Besides, can I use charming in vscode jupyter and render them ?

**Comment by yuankunzhang** [comment]
Hi @wangjiawen2013 , thanks for the feedback!

At the moment I don't have a concrete roadmap, but I do have something in my mind: adding more documentations being the most important one and adding support for 3D charts being another. Making it accessible in Jupyter is an interesting (and a very useful) idea, I'll find some time to look into it.

Btw you can use the `Chart.to_string()` method to dump a chart definition into a JSON string. And any form of contributions is welcome!

**Comment by locke4** [comment]
This is a fantastic library by the way, I too found plotters and plotly weren't as useful as this library. I'm testing it out with WASM and Yew. One suggestion for the roadmap, I'd be really keen to either see a histogram series type or an option with bar charts to set the width in relative interval % rather than absolute screen pixels. I'm currently calculating the pixels based on the chart width.

**Comment by wangjiawen2013** [comment]
It's also helpful to add some examples on how to work with external dataset, such as excel, csv, polars dataframe , because some user's data are stored in excel and csv. Vega-lite altair supports pandas dataframe and is more friendly to users.

**Comment by humphreylee** [comment]
Is the project still alive?

**Comment by wangjiawen2013** [comment]
I'm afraid not. There is a very lack of tutorials and practical chart types. 
But there is a good new, the rust plotly project is re-vived.

**Comment by humphreylee** [comment]
Thanks. Will check it out.

**Comment by humphreylee** [comment]
> It's also helpful to add some examples on how to work with external dataset, such as excel, csv, polars dataframe , because some user's data are stored in excel and csv. Vega-lite altair supports pandas dataframe and is more friendly to users.

A macro has been [implemented](https://github.com/yuankunzhang/charming/pull/100) to convert vectors into an EChart-compatible dataframe, enabling the use of various data formats within Charming.

**Comment by LukaOber** [comment]
> It's also helpful to add some examples on how to work with external dataset, such as excel, csv, polars dataframe , because some user's data are stored in excel and csv. Vega-lite altair supports pandas dataframe and is more friendly to users.

I don't think a general approach to convert csv or excel data is possible. You would still need to manually read the data from your source file and convert it to a compatible dataframe for charming. The new dz! macro will remove a lot of the boilerplate code for this.

**Comment by LukaOber** [comment]
> Is the project still alive?

For anyone finding this in the future, it is. We are currently trying to add more fields to allow more customizable charts, adding tests to check correct serialization and also trying to simplify some internal stuff.

Most issues and pull requests get an answer or suggestion in the first few hours.

**Comment by humphreylee** [comment]
> > Is the project still alive?
> 
> For anyone finding this in the future, it is. We are currently trying to add more fields to allow more customizable charts, adding tests to check correct serialization and also trying to simplify some internal stuff.
> 
> Most issues and pull requests get an answer or suggestion in the first few hours.

I like this library so much that I've already contributed three PRs to help move the project forward.

**Comment by maqui7295** [comment]
Hi, thanks for this awesome project. 
Is there any plan to support echarts v6? Thanks.

**Comment by LukaOber** [comment]
Yes, we will switch to it in the future. I don't know when I will get to it though. What features do you need/want from the new version?

**Comment by LukaOber** [comment]
More maintainers are always welcome. Please don't hesitate to open discussions or issues regarding any questions.
